src/main/java/org/jetbrains/bsp/bazel/server/BazelBspServer.java

```diff
@@ -77,7 +81,7 @@ private void integrateBsp(BspIntegrationData bspIntegrationData) {
             .traceMessages(bspIntegrationData.getTraceWriter())
             .setOutput(bspIntegrationData.getStdout())
             .setInput(bspIntegrationData.getStdin())
-            .setLocalService(buildServer)
+            .setLocalService(aggregateBuildServerImpl)
             .setRemoteInterface(BuildClient.class)
             .setExecutorService(bspIntegrationData.getExecutor())
```

Comment:
After some research I was able to find why the `setLocalServices` call probably didn't work. For a collection of local services, one should also set  a collection of remote interfaces using `setRemoteInterfaces`. This call can be seen in use [here](https://github.com/eclipse/lsp4j/blob/23f42a41288bb45ca088ca4cb75e2c9a70259e13/org.eclipse.lsp4j.jsonrpc/src/main/java/org/eclipse/lsp4j/jsonrpc/Launcher.java#L206). Lmk how that works out!

Reply:
What exactly should I pass to ``setRemoteInterfaces`` then?

Reply:
Disregard this :sweat: , I thought for a second that a client to support extension methods also needed to implement some classes which is not true